### DISCUSSION TITLE
Ellipsoid arbitrary orientation

### DISCUSSION THREAD

**Discussion by alpha027** [pyvista/pyvista#1650] [Q&A] [answered]
Hi,

I am writing regarding the [parametric ellipsoid](https://github.com/pyvista/pyvista/blob/1415b2a1b443ea0f5c6ed5851652b74b1bda3017/pyvista/utilities/parametric_objects.py#L391).
As mentioned in the documentation and the code, the method takes three radii (xradius, yradius, zradius). However, it takes only one direction -I assume it is the direction of the first axis with length corresponding to xradius-. If the radii of the ellipsoid are all distinct (the case of exact ellipsoid and not a spheroid), then we can't plot an ellipsoid with an arbitrary orientation in space. 
Am I wrong and it is actually possible to plot arbitrary ellipsoid ? if that's the case please let me know how. If it's not, I would love to have this possibility available.

![Screenshot from 2021-09-07 15-00-54](https://user-images.githubusercontent.com/27587257/132351392-4fdbacd2-bbba-49e0-8e38-cc3e6822c724.png)

With the current function of parametric ellipsoid, it is not possible to plot the ellipsoids that are obtained by rotating the one in the image according to the rotation axis [1,1,1].

Best.

Anas

**Comment by akaszynski** [comment, marked as answer]
I'd rotate the ellipsoid after creating it using [`rotate_x`](https://docs.pyvista.org/core/dataset.html?highlight=rotate_x#pyvista.DataSet.rotate_x), [`rotate_y`](https://docs.pyvista.org/core/dataset.html?highlight=rotate_x#pyvista.DataSet.rotate_y), and [`rotate_z`](https://docs.pyvista.org/core/dataset.html?highlight=rotate_x#pyvista.DataSet.rotate_z).

**Comment by alpha027** [comment]
Thanks that solved my problem.

A minor suggestion would be to have another way of constructing ellipsoid object from its center vector and three vectors of axes -this could be more straightforward-.

**Reply by akaszynski** [reply]
That's a good idea.  Since I'm swamped at the moment for work, would you be interested in implementing it and suggesting a PR for it?  The pertinent file is [parametric_objects.py](https://github.com/pyvista/pyvista/blob/main/pyvista/utilities/parametric_objects.py)

**Comment by alpha027** [comment]
Hi, thanks for the reply, I'll take look into it, see what I can do and reach to you back again.
